docs/investigation/multi-cluster/README.md

```diff
@@ -0,0 +1,279 @@
+                - name: kubernetes
+                  optional: true
+            steps:
+              - - name: create-target-cluster
+                  capact-when: kubernetes == nil
+                  capact-action: k8s.deploy
+              - - name: postgresql-install
+                  capact-action: postgresql.install
+                  capact-policy:
+                    rules:
+                      - interface:
+                          path: postgresql.install
+                        oneOf:
+                          - implementationConstraints:
+                              requires:
+                                - path: "cap.core.type.platform.kubernetes"
+                            inject:
+                              requiredTypeInstances:
+                                - artifact: "{{steps.create-target-cluster.outputs.artifacts.kubernetes}}"
+```
+
+Capact user or admin can provide already existing Kubernetes cluster via Policy:
+
+```yaml
+rules:
+  - interface: 
+      path: "cap.interface.database.postgresql.bootstrap-cluster"
+    oneOf:
+      - implementationConstraints: # In first place, find and use an Implementation which:
+          path: "cap.implementation.postgresql.bootstrap-cluster"
+        inject:
+          additionalTypeInstances: # Injects additional TypeInstances for the Implementation
+            - name: kubernetes # Name must match one of the parameter defined under `additionalInput.typeInstances` in the Implementation
+              id: 0b6dba9a-d111-419d-b236-357cf0e8603a
+```
+
+### Use pre-existing (registered) cluster
+
+Example of the Implementation manifest that installs the PostgreSQL on Kubernetes platform:
+
+```yaml
+kind: Implementation
+metadata:
+  # ...
+  displayName: Install PostgreSQL database
+spec:
+  # ...
+  requires:
+    cap.core.type.platform:
+      oneOf:
+        - name: kubernetes
+          revision: 0.1.0
+
+  action:
+    runnerInterface: argo.run
+    args:
+      workflow:
+        entrypoint: postgres-install
+        templates:
+          - name: postgres-install
+            steps:
+              - - name: helm-install
+                  # Argo renderer mutates the container definition to ensure that it's created on a given target cluster. 
+                  # It's done based on injected `cap.core.type.platform.kubernetes` TypeInstance
+                  capact-action: helm.install
+```
+
+If necessary we can introduce the `capact-target` property which can be used to mark a given step to be executed on a given cluster. Without `capact-target` step is not mutated, so it's executed where Argo was installed:
+
+```yaml
+# ...
+  requires:
+    cap.core.type.platform:
+      oneOf:
+        - name: kubernetes
+          alias: k8s
+          revision: 0.1.0
+
+# ...
+            steps:
+              - - name: helm-install
+                  capact-action: helm.install
+                  capact-target: k8s # based on alias in requires section 
```

Comment:
So effectively `capact-target` would modify the step, adding `capact-policy` property with policy injecting a given Kubeconfig (the one preregistered in cluster or provided as `capact-target`), correcto?

Reply:
discussed f2f